packages/expo-file-system/android/src/main/java/expo/modules/filesystem/ExpokitFilePermissionChecker.java

```diff
@@ -0,0 +1,58 @@
+package expo.modules.filesystem;
+
+import android.content.Context;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+
+import expo.core.interfaces.InternalModule;
+import main.FilePermissionChecker;
+import main.Permission;
+
+public class ExpokitFilePermissionChecker implements FilePermissionChecker, InternalModule {
+  @Override
+  public EnumSet<Permission> getInfo(Context context, String path) {
+    try {
+      path = new File(path).getCanonicalPath();
+      String filesDir = context.getFilesDir().getCanonicalPath();
+      if (path.startsWith(filesDir + "/")) {
+        return EnumSet.of(Permission.READ, Permission.WRITE);
+      }
+      if (filesDir.equals(path)) {
+        return EnumSet.of(Permission.READ, Permission.WRITE);
+      }
+      String cacheDir = context.getCacheDir().getCanonicalPath();
+      if (path.startsWith(cacheDir + "/")) {
+        return EnumSet.of(Permission.READ, Permission.WRITE);
+      }
+      if (cacheDir.equals(path)) {
+        return EnumSet.of(Permission.READ, Permission.WRITE);
+      }
+    } catch (IOException e) {
+      return EnumSet.noneOf(Permission.class);
+    }
+    return checkForPath(path);
+  }
+
+  protected EnumSet<Permission> checkForPath(String path) {
```

Comment:
Is it a method only to be used when the `path` is out of the `filesDir` and `cacheDir`? If so, I think `checkForPath` doesn't really describe it well. I would either contain all the logic in one `getInfo` method (which maybe should be named `getPermissions`) or I would create three separate methods:
- `getPermissions` with `if` inside discerning between
    -  `getPermissionsForInternalStorage(Context context, String path)`
    -  `getPermissionsForExternalStorage(Context context, String path)`